backend/crypt/cipher.go

```diff
@@ -114,22 +117,65 @@ func (mode NameEncryptionMode) String() (out string) {
 	return out
 }
 
-// Cipher defines an encoding and decoding cipher for the crypt backend
+// StrEncoding is encoding methods dealing with strings
+type StrEncoding interface {
+	EncodeToString(src []byte) string
+	DecodeString(s string) ([]byte, error)
+}
+
+type CaseInsensitiveBase32Encoding struct {
```

Comment:
This seems over complicated to me....

I'd suggest you define this (note lower case so we don't make a public symbol)

```go
type caseInsensitiveBase32Encoding struct {}
```

Then define the two methods

```
func (_ caseInsensitiveBase32Encoding) EncodeToString(src []byte) string {
// contents the same as encodeFileName in original source
}
func (_ caseInsensitiveBase32Encoding) DecodeString(s string) ([]byte, error) {
// contents the same as decodeFileName in original source
}
```

And finally check it satisfies the interface

```
var _ StrEncoding = caseInsensitiveBase32Encoding{}
```

Reply:
Indeed this change is modifying codes that already work. I reverted the base32 implementation in 79817be.